File: api/src/main/java/io/github/mh321productions/serverapi/module/log/LogWrapper.java
package io.github.mh321productions.serverapi.module.log;

import java.io.File;
import java.io.FileNotFoundException;
import java.util.logging.Level;
import java.util.logging.Logger;

import io.github.mh321productions.serverapi.Main;
import io.github.mh321productions.serverapi.api.SubPlugin;
import io.github.mh321productions.serverapi.util.logging.PixelLogHandler;

/**
 * Eine Wrapper-Klasse für das {@link LogModule Logging-Modul}
 * @author 321Productions
 */
public class LogWrapper {
	
	public SubPlugin sub;
	public Logger log;
	public PixelLogHandler handler;
	public File logFile;
	
	public LogWrapper(Main plugin, SubPlugin sub) {
		this.sub = sub;
		log = sub.getLogger();
		logFile = plugin.filesystem.getLogFile(sub);
	}
	
	/**
	 * Initialisiert das Umleiten des Loggers
	 * @return Ob das Umleiten erfolgreich war
	 */
	public boolean init() {
		log.info("Starte API Logger");
		
		try {
			handler = new PixelLogHandler(logFile);
			log.addHandler(handler);
			log.info("API Logger gestartet");
		} catch (FileNotFoundException e) {
			//e.printStackTrace();
			log.log(Level.SEVERE, "Der API Logger konnte nicht gestartet werden:", e);
			return false;
		}
		
		return true;
	}
	
	public void end() {
		log.info("Stoppe API Logger");
		
		handler.close();
		log.removeHandler(handler);
		
		log.info("API Logger gestoppt");
	}
}

File: src/main/kotlin/io/github/mh321productions/serverapi/module/log/LogWrapper.kt
package io.github.mh321productions.serverapi.module.log

import io.github.mh321productions.serverapi.Main
import io.github.mh321productions.serverapi.api.SubPlugin
import io.github.mh321productions.serverapi.util.logging.PixelLogHandler
import java.io.File
import java.io.IOException
import java.util.logging.Level
import java.util.logging.Logger

/**
 * Eine Wrapper-Klasse für das [Logging-Modul][LogModule]
 * @author 321Productions
 */
class LogWrapper(plugin: Main, val sub: SubPlugin) {

    val log: Logger = sub.logger
    val handler: PixelLogHandler
    private val logFile: File = plugin.filesystem.getLogFile(sub)

    init {
        log.info("Starte API Logger")

        handler = try {
            val handler = PixelLogHandler(logFile)
            log.addHandler(handler)
            log.info("API Logger gestartet")
            handler
        } catch (e: IOException) {
            log.log(Level.SEVERE, "Der API Logger konnte nicht gestartet werden:", e)
            throw e
        }
    }

    fun end() {
        log.info("Stoppe API Logger")

        handler.close()
        log.removeHandler(handler)

        log.info("API Logger gestoppt")
    }
}
